src/react/RegisterDevicePage.tsx

```diff
@@ -21,11 +21,19 @@
 import React, { useCallback, useMemo, useState } from 'react';
 import { v4 as uuidv4, v5 as uuidv5 } from 'uuid';
 import { Button, Col, Form, Modal, Spinner, Table } from 'react-bootstrap';
+import AstarteClient from 'astarte-client';
+import type { AstarteDevice } from 'astarte-client';
 
 import SingleCardPage from './ui/SingleCardPage';
 import { byteArrayToUrlSafeBase64, urlSafeBase64ToByteArray } from './Base64';
 import { useAlerts } from './AlertManager';
 
+interface InterfaceDescriptor {
```

Comment:
this is already defined in the astarte client. Import it from there?

Reply:
I wasn't sure about defining a proper type to export from `astarte-client`. However `AstarteInterfaceDescriptor` is a well defined entity and can be useful; i followed your advice